scripts/irods/test/test_iadmin.py

```diff
@@ -1433,3 +1433,67 @@ def test_unused_r_data_main_columns(self):
         self.admin.assert_icommand(['iadmin', 'rsq', query_name])
         self.admin.assert_icommand(['irm', '-f', test_file])
         os.unlink(test_file)
+
+class Test_Issue3862(resource_suite.ResourceBase, unittest.TestCase):
+
+    def setUp(self):
+        super(Test_Issue3862, self).setUp()
+
+        output = commands.getstatusoutput("hostname")
+        hostname = output[1]
+
+        # =-=-=-=-=-=-=-
+        # STANDUP
+        self.admin.assert_icommand("iadmin mkresc repl replication", 'STDOUT_SINGLELINE', "Creating")
+
+        self.admin.assert_icommand("iadmin mkresc leaf_a unixfilesystem " + hostname +
+                                   ":/tmp/irods/pydevtest_leaf_a", 'STDOUT_SINGLELINE', "Creating")  # unix
+        self.admin.assert_icommand("iadmin mkresc leaf_b unixfilesystem " + hostname +
+                                   ":/tmp/irods/pydevtest_leaf_b", 'STDOUT_SINGLELINE', "Creating")  # unix
+
+        # =-=-=-=-=-=-=-
+        # place data into the leaf_a
+        test_dir = 'issue3862'
+        num_files = 400
+        if not os.path.isdir(test_dir):
+            os.mkdir(test_dir)
+        for i in range(num_files):
+            filename = test_dir + '/file_' + str(i)
+            lib.create_local_testfile(filename)
+
+        with session.make_session_for_existing_admin() as admin_session:
+
+            # add files to leaf_a
+            admin_session.run_icommand(['iput', '-r', '-R', 'leaf_a', test_dir])
+
+            # now connect leaves to repl
+            admin_session.run_icommand(['iadmin', 'addchildtoresc', 'repl', 'leaf_a'])
+            admin_session.run_icommand(['iadmin', 'addchildtoresc', 'repl', 'leaf_b'])
+
+    def tearDown(self):
+
+        super(Test_Issue3862, self).tearDown()
+
+        test_dir = 'issue3862'
+
+        with session.make_session_for_existing_admin() as admin_session:
+
+            admin_session.run_icommand(['irm', '-rf', test_dir])
+
+            admin_session.run_icommand(['iadmin', 'rmchildfromresc', 'repl', 'leaf_a'])
+            admin_session.run_icommand(['iadmin', 'rmchildfromresc', 'repl', 'leaf_b'])
+
+            admin_session.run_icommand(['iadmin', 'rmresc', 'leaf_b'])
+            admin_session.run_icommand(['iadmin', 'rmresc', 'leaf_a'])
+            admin_session.run_icommand(['iadmin', 'rmresc', 'repl'])
+
+    # Issue 3862:  test that CAT_STATEMENT_TABLE_FULL is not encountered during parallel rebalance
+    def test_rebalance__ticket_3862(self):
+
+        # =-=-=-=-=-=-=-
+        # call two separate rebalances, the first in background
+        # prior to fix the second would generate a CAT_STATEMENT_TABLE_FULL error
+        #self.admin.assert_icommand("iadmin modresc repl rebalance&")
```

Comment:
please remove this commented line.

if everything else passes in 4-2-stable, we'll merge.